juriscraper/scotus/scotus_docket_report_htm.py

```diff
@@ -0,0 +1,451 @@
+                    ".//tr[td[1][contains(normalize-space(.), '~~~Date~~~')] "
+                    "and td[2][contains(normalize-space(.), 'Proceedings')]]"
+                )
+            ),
+            None,
+        )
+        if header_tr is None:
+            return []
+
+        entries = []
+
+        # Only iterate rows after the header within the same table
+        for tr in header_tr.itersiblings(tag="tr"):
+            tds = tr.xpath("./td")
+            if not tds:
+                continue
+            date_str = self._clean_whitespace(tds[0].text_content())
+            desc_td = tds[1]
+
+            # Build the entry
+            entries.append(
+                self._build_docket_entry(
+                    date_str=date_str,
+                    description_td=desc_td,
+                    attachment_path=".//a[@href]",
+                )
+            )
+
+        return entries
+
+    def _build_htm_attorney(
+        self,
+        current_attorney: Optional[dict[str, Any]],
+        current_type: Optional[str],
+        current_party_name: str,
+        parties_by_key: dict[tuple[str, str], list],
+    ) -> Optional[dict[str, Any]]:
+        """Complete an attorney dict from the HTM Contacts table.
+
+        :param current_attorney: The in-progress attorney dict with initial values.
+        :param current_type: Current section type ('Petitioner', 'Respondent', or 'Other').
+        :param current_party_name: The party name that this attorney belongs to.
+        :param parties_by_key: parties by type-party for grouping.
+        :return: None. The current_attorney is modified in place.
+        """
+        if current_attorney is None:
+            return None
+
+        lines = current_attorney.pop("_raw_lines", [])
+        email = current_attorney.pop("_email", None)
+
+        # Filter lines: remove lines with IDs like "#1098260" and inline email if present.
+        filtered_lines = []
+        for ln in lines:
+            if email is None:
+                m = self.EMAIL_RE.search(ln)
+                if m:
+                    email = m.group(0)
+                    continue
+            if self.ID_RE.search(ln):
+                continue
+            if ln:
+                filtered_lines.append(ln)
+
+        title, start_add_idx = self._parse_address_title(filtered_lines)
+        partial_address = self._parse_address_lines(
+            filtered_lines, start_add_idx
+        )
+
+        addr_lines = partial_address.address_lines
+        current_attorney.update(
+            {
+                "title": title,
+                "address": ", ".join(addr_lines) if addr_lines else None,
+                "city": partial_address.city,
+                "state": partial_address.state,
+                "zip": partial_address.zip_code,
+                "email": email,
+            }
+        )
+        type_key = (current_type or "Other", current_party_name)
+        parties_by_key[type_key].append(current_attorney)
```

Comment:
The `_build_htm_attorney` method has a responsibility that isn't immediately obvious. The line `parties_by_key[type_key].append(current_attorney)` updates the main parties list, which is an action that I'd expect to see in the calling function (`parties`).

Consider refactoring this to have `_build_htm_attorney` only return the completed attorney dictionary. The calling function can then handle the appending. This separation of concerns would make the code's behavior more predictable and easier to debug.

Reply:
You're right. It's better to keep the update of `parties_by_key` outside of this method. I've applied the suggested refactor.